src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -137,6 +141,53 @@ public Channel createChannel(Course course, Channel channel, Optional<User> crea
         return savedChannel;
     }
 
+    /**
+     * Register users to the newly created channel
+     *
+     * @param addAllStudents        if true, all students of the course will be added to the channel
+     * @param addAllTutors          if true, all tutors of the course will be added to the channel
+     * @param addAllInstructors     if true, all instructors of the course will be added to the channel
+     * @param usersLoginsToRegister the logins of the users to register to the channel
+     * @param course                the course to create the channel for
+     * @param channel               the channel to create
+     * @return all users that were registered to the channel
+     */
+    public Set<User> registerUsersToChannel(boolean addAllStudents, boolean addAllTutors, boolean addAllInstructors, List<String> usersLoginsToRegister, Course course,
+            Channel channel) {
+        Set<User> usersToRegister = new HashSet<>();
+        usersToRegister.addAll(conversationService.findUsersInDatabase(course, addAllStudents, addAllTutors, addAllInstructors));
+        usersToRegister.addAll(conversationService.findUsersInDatabase(usersLoginsToRegister));
+        conversationService.registerUsersToConversation(course, usersToRegister, channel, Optional.empty());
+        return usersToRegister;
+    }
+
+    /**
+     * Add user to default channels of courses with the same group asynchronously. This is used when a user is added to a group.
+     *
+     * @param userToAddToGroup the user to be added
+     * @param group            the group of the user
+     * @param role             the role of the user
+     */
+    @Async
+    public void registerUserToDefaultChannels(User userToAddToGroup, String group, Role role) {
+        final Set<String> channelNames = Arrays.stream(DefaultChannelType.values()).map(DefaultChannelType::getName).collect(Collectors.toSet());
+
+        List<Course> courses = switch (role) {
+            case STUDENT -> courseRepository.findCoursesByStudentGroupName(group);
+            case TEACHING_ASSISTANT -> courseRepository.findCoursesByTeachingAssistantGroupName(group);
+            case INSTRUCTOR -> courseRepository.findCoursesByInstructorGroupName(group);
+            default -> List.of();
+        };
```

Comment:
This should probably be extracted to the courseRepository as a default method

Reply:
I will probably extract this in #6513 PR.  I would leave as it is and merge this PR